include/sampleflow/consumers/last_sample.h

```diff
@@ -79,35 +199,190 @@ namespace SampleFlow
-         * The value of the last sample seen by consume().
+         * A variable that describes the left end points of each of the
+         * intervals that make up each bin. The vector contains one additional
+         * element that denotes the right end point of the last interval.
+         */
+        std::vector<double> interval_points;
+
+        /**
+         * A vector storing the number of samples so far encountered in each
+         * of the bins of the histogram.
+         */
+        std::vector<types::sample_index> bins;
+
+        /**
+         * For a given `value`, compute the number of the bin it lies
+         * in, taking into account the way the bins subdivide the
+         * range for which a histogram is to be computed.
+         *
+         * If the given value lies to the left of the left-most interval,
+         * or the right of the right-most interval, then this function will
+         * abort.
          */
-        InputType last_sample;
+        unsigned int bin_number (const double value) const;
     };
 
 
 
+    template <typename InputType>
+    Histogram<InputType>::
+    Histogram (const double min_value,
+               const double max_value,
+               const unsigned int n_bins)
+      :
+      interval_points(n_bins+1),
+      bins (n_bins)
+    {
+      assert (min_value < max_value);
+
+      // Set up the break points between the bins:
+      const double delta = (max_value - min_value) / (n_bins);
+      for (unsigned int bin=0; bin<n_bins; ++bin)
+        interval_points[bin] = min_value + bin*delta;
+
+      // And add the past-the-end interval's left end point as well:
+      interval_points[n_bins] = max_value;
+    }
+
+
+    template <typename InputType>
+    Histogram<InputType>::
+    Histogram (const double min_pre_value,
+               const double max_pre_value,
+               const unsigned int n_bins,
+               const std::function<double (const double)> &f)
+      :
+      interval_points(n_bins+1),
+      bins (n_bins)
+    {
+      assert (min_pre_value < max_pre_value);
+
+      // Set up the break points between the bins:
+      const double delta = (max_pre_value - min_pre_value) / (n_bins);
+      for (unsigned int bin=0; bin<n_bins; ++bin)
+        interval_points[bin] = f(min_pre_value + bin*delta);
+
+      // And add the past-the-end interval's left end point as well:
+      interval_points[n_bins] = f(max_pre_value);
+
+
+      // Double check that the mapping used was indeed strictly
+      // increasing:
+      for (unsigned int bin=0; bin<n_bins; ++bin)
+        {
+          assert (interval_points[bin] < interval_points[bin+1]);
+        }
+    }
+
+
+
+    template <typename InputType>
+    Histogram<InputType>::
+    Histogram (const Histogram<InputType> &o)
+      :
+      interval_points(o.interval_points),
+      bins (o.bins)
+    {}
+
+
+
     template <typename InputType>
     void
+<<<<<<< HEAD
+<<<<<<< HEAD
     LastSample<InputType>::
+=======
```

Comment:
This also shouldn't happen -- you accidentally committed conflict markers.

Reply:
fixed